### PR TITLE
Add share button to proposal details

### PR DESCRIPTION
**Pull request by dkent600**
Resolves https://daostack.tpondemand.com/RestUI/Board.aspx#page=board/5209716961861964288&appConfig=eyJhY2lkIjoiQjgzMTMzNDczNzlCMUI5QUE0RUE1NUVEOUQyQzdFNkIifQ==&boardPopup=userstory/839/silent

### CODE REVIEW

**Comment by dkent600** [comment]
There are still minor questions about content, for @A-Zak  but the code is ready for review.

**Comment by dkent600** [comment]
Ready for review

File: src/components/Shared/SocialShareModal.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  closeHandler: (event: any) => void;
+  url: string;
+}
+
+export default class SocialShareModal extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.selectTwitter = this.selectTwitter.bind(this);
```
Comment (tibetsprague) [inline src/components/Shared/SocialShareModal.tsx]: FYI there is another way around having to do this binding, using arrow functions like this: private showCopiedFeedback = (event: any) => {...} or if you need to pass a variable to the handler showCopiedFeedback = (param: number) => (event: any) => {...}

https://medium.com/silesis/handle-events-in-react-with-arrow-functions-ede88184bbb

File: src/components/Shared/SocialShareModal.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    this.selectTelegram = this.selectTelegram.bind(this);
+    this.copyUrl = this.copyUrl.bind(this);
+
+    this.state = {
+      showCopiedFeedback: false,
+    };
+  }
+
+  private showCopiedFeedback(): void {
+    this.setState({ showCopiedFeedback: true });
```
Comment (tibetsprague) [inline src/components/Shared/SocialShareModal.tsx]: elsewhere we use showNotification for this, and those notifications are supposed to go away on their own, but that stopped working at some point... would be good to still use that unified notification system though

Comment (dkent600) [reply src/components/Shared/SocialShareModal.tsx]: The spec specifically asked for this design, not a notification.

**Pull request merged**